components/dashboard/src/Menu.tsx

```diff
@@ -125,20 +126,18 @@ export default function Menu() {
     ]);
     const isAdminUI = inResource(window.location.pathname, ["admin"]);
 
-    const [teamMembers, setTeamMembers] = useState<Record<string, TeamMemberInfo[]>>({});
+    const [teamMembers, setTeamMembers] = useState<Record<string, TeamMember[]>>({});
 
     useEffect(() => {
         if (!teams) {
             return;
         }
         (async () => {
-            const members: Record<string, TeamMemberInfo[]> = {};
+            const members: Record<string, TeamMember[]> = {};
             await Promise.all(
                 teams.map(async (team) => {
                     try {
-                        members[team.id] = publicApiTeamMembersToProtocol(
-                            (await teamsService.getTeam({ teamId: team!.id })).team?.members || [],
-                        );
+                        members[team.id] = (await teamsService.getTeam({ teamId: team!.id })).team?.members || [];
```

Comment:
Since we already have the team we don't need to fetch it again here

Reply:
Makes sense. I'd like to make this change first, which is just one-to-one change of stopping the conversion and we can apply this optimisation on top.